New Database + Macro

UltimateNyn opened the pull request:
This code uses the new Database + adds functions to get the config from the DB instead of using a raw struct. Also the repetitive
```rs
match foo {
    Ok(val) => val,
    Err(why) => {
        log::error!("error [...]: {}", why);
        return;
    },
}
```

was replaced with a macro (``ttc_unwrap``).
Please note that this should be tested a bit more before getting merged.

Other changes:
- removed ``create_webhooks``
- remove ``-w`` argument for now

For a detailed list look in the commit history

kilea-chan committed:
Redesigned database

kilea-chan committed:
Added emoji cache tables

kilea-chan committed:
Restructured ttc_config relations

kilea-chan committed:
Fixed wrongly generated SQL

kilea-chan committed:
Renamed counted_emoji to harold_emoji

kilea-chan committed:
Populate first 500 entries of the message cache

kilea-chan committed:
Create view of config table

kilea-chan committed:
Merge branch 'develop' into develop

kilea-chan committed:
Explicitly name columns

UltimateNyn committed:
Use DB to identify Harold emojis

kilea-chan committed:
Permissions should be inherited

UltimateNyn committed:
Fix conveyance blacklist for deleted messages

Kirottu committed:
The Config struct is no more, but everything thinks it is. It has been replaced with config key implementations in the Data struct

kilea-chan committed:
Fixed too small channel_id type

UltimateNyn committed:
Merge branch 'kiro/new-db' into new-db-v2

UltimateNyn committed:
Remove -w option for now

UltimateNyn committed:
Fix conveyance sql rename

UltimateNyn committed:
Make new config work

UltimateNyn committed:
Replace repetitive match Error with macro

UltimateNyn committed:
remove deprecated create_webhooks

UltimateNyn committed:
Use DB for selfroles

UltimateNyn committed:
add emojis from DB to selfroles

UltimateNyn committed:
fix possible panic with duration overflows

UltimateNyn committed:
fix 0 second duration bug

UltimateNyn committed:
optimize create_selfroles to have O(n) runtime

UltimateNyn committed:
remove old get function in config

UltimateNyn committed:
use colors from DB and add default colors

UltimateNyn committed:
fix missing semicolon in SQL file

Kirottu reviewed (changes requested):
Looks pretty weird and hacky in places, the comments explain it better.

UltimateNyn committed:
fix comments from the PR except reply + sql macro

Kirottu committed:
Here is a fun "little" commit to do some fixes

- Removed the weird reply function mess
- Moved color and config key functions into their own structs
- Created `ContextExt` trait for `send_simple` and `send_embed` shortcut functions for the Context
- Created `Readable` trait for more user facing formatting
- Moved types and traits into their respective subfolders, more of them should probably be moved from various source files
- Probably something else I forgot about

Kirottu committed:
Finally removed the crazy string date parsing conversion thing and disabled the useless (at least with this implementation) check_duration

Kirottu committed:
Removed unused #[macro_export] in colors.rs

Kirottu reviewed (commented):
Seems to be looking pretty good, old commented code needs to be removed and proper testing needs to commence before merging.

UltimateNyn committed:
fix embed_color

The old macro has a typo which resulted in that every query would return
no results. Furthermore ``embed_type`` was made unique to prevent
multiple values for the same embed type

UltimateNyn committed:
fix segfault under x86-64 musl

Use rustls-tls for reqwest. The other way to prevent the crash would
have been to set ``RUSTFLAGS`` to ``-Ctarget-feature=-crt-static``

UltimateNyn committed:
remove old commented out code

UltimateNyn committed:
Merge remote-tracking branch 'fork/new-db-v2' into new-db-v2

Kirottu reviewed (approved):
Alright, looks good.
